botnim/kb/manager.py

```diff
@@ -0,0 +1,179 @@
+        
+        This is useful when:
+        - Content is manually curated
+        - Information is naturally split into distinct files
+        - Content comes from multiple sources
+        
+        Args:
+            context_config: Configuration dictionary containing:
+                - split: Path to directory containing split markdown files
+                
+        Returns:
+            List of tuples (filename, file_handle, content_type) for each markdown file
+        
+        Example config:
+            context:
+              - name: Manual Knowledge
+                split: manual_kb    # Directory containing .md files
+        """
+        dir_path = self.config_dir / context_config['split']
+        
+        if not dir_path.exists():
+            logger.warning(f"Split directory not found: {dir_path}")
+            return []
+
+        documents = []
+        for file_path in sorted(dir_path.glob('*.md')):
+            if file_path.read_text().strip():  # Skip empty files
+                env_filename = self._add_environment_suffix(file_path.name)
+                documents.append((
+                    env_filename,
+                    file_path.open('rb'),
+                    'text/markdown'
+                ))
+            else:
+                logger.debug(f'Skipping empty file: {file_path}')
+        
+        return documents
+
+    def collect_documents(self, context_config: dict) -> List[Union[BinaryIO, Tuple[str, BinaryIO, str]]]:
+        """Collect documents from all configured sources
+        
+        Supports multiple source types:
+        1. Spreadsheet source (type: spreadsheet)
+        2. Split files (split: directory_path)
+        3. Regular files (files: glob_pattern)
+        
+        Args:
+            context_config: Configuration dictionary specifying the source type and location
+            
+        Returns:
+            List of documents in the format required by the vector store
+        """
+        documents = []
+        
+        # Handle spreadsheet source
+        if context_config.get('type') == 'spreadsheet' and 'source' in context_config:
+            from .download_sources import download_and_convert_spreadsheet
+            split_path = (
+                self.config_dir / context_config['split'] 
+                if 'split' in context_config 
+                else self.config_dir / f"{context_config['name']}_split"
+            )
+            documents.extend([
+                (self._add_environment_suffix(filename), file_obj, content_type)
+                for filename, file_obj, content_type 
+                in download_and_convert_spreadsheet(
+                    context_config['source'], 
+                    split_path, 
+                    context_config['name']
+                )
+            ])
+        
+        # Handle split files
+        elif 'split' in context_config:
+            documents.extend(self._process_split_file(context_config))
```

Comment:
can simply return here without extend (same for all branches)

Reply:
ok